cdap-api/src/main/java/io/cdap/cdap/api/macro/MacroEvaluator.java

```diff
@@ -46,11 +51,34 @@
 
   /**
    * Use the macro function and call the function with provided arguments,
-   * function uses the arguments and returns the evaluated response.
+   * function uses the arguments and returns the evaluated response as a string.
+   *
    * @param macroFunction macro function that has to be called
    * @param arguments arguments that will be passed to the macro function
    * @return value returned by macro function
    * @throws InvalidMacroException if macroFunction is not supported
    */
   String evaluate(String macroFunction, String... arguments) throws InvalidMacroException;
+
+  /**
+   * Use the macro function and call the function with provided arguments,
+   * function uses the arguments and returns the evaluated response as a map.
```

Comment:
Should mention that the map values can also contain macros and will be evaluated, but map keys cannot contain macros.